commands/operator-sdk/cmd/generate/openapi.go

```diff
@@ -69,13 +68,23 @@ func openAPIFunc(cmd *cobra.Command, args []string) error {
 		return fmt.Errorf("command %s doesn't accept any arguments", cmd.CommandPath())
 	}
 
-	return OpenAPIGen()
+	return OpenAPIGen(headerFile)
 }
 
-// OpenAPIGen generates OpenAPI validation specs for all CRD's in dirs.
-func OpenAPIGen() error {
+// OpenAPIGen generates OpenAPI validation specs for all CRD's in dirs. hf is
+// a path to a header file containing text to add to generated files.
+func OpenAPIGen(hf string) error {
 	projutil.MustInProjectRoot()
 
+	hft, rm, err := genutil.GetHeaderFileIfEmpty(hf)
```

Comment:
Nit: totally a style thing, but this seems like a situation where it would be useful to pass the function that uses the header file in, rather than expect the caller to handle the cleanup.

```go
func WithHeaderFile(hf string, f func(string) error) error {
    var err error
    if hf == "" {
        hf, err = createEmptyHeaderFile()
        if err != nil {
            return err
        }
        defer func() {
	    if err = os.RemoveAll(hf); err != nil {
	        log.Error(err)
	    }
	}()
    }
    return f(hf)
}

Reply:
Much better idea, thanks!